### PR TITLE
Fix yaw not matching client yaw

### PR DESCRIPTION
**Pull request by jivvy**
Saw this bug ( #121 ) and thought it might be a good first issue.

I'm new to rust and contributing in general, so please guide me.

This seemed to fix the issue, but it might not apply to all cases, only for C2S packets regarding pos & yaw and entities

I'm also not sure what the best way of applying the wrap to when you set the yaw, though.

### CODE REVIEW

**Comment by jivvy** [comment]
I put a print statement in where the C2S packets are handled:
```rust
C2sPlayPacket::SetPlayerPositionAndRotation(p) => {
    println!("yaw: {}", p.yaw);
    if self.pending_teleports == 0 {
        self.position = p.position;
        self.yaw = wrap_yaw(p.yaw);
        self.pitch = p.pitch;

        self.events.push_back(ClientEvent::MovePositionAndRotation {
            position: p.position,
            yaw: p.yaw,
            pitch: p.pitch,
            on_ground: p.on_ground,
        });
    }
}
```
And interpreted this as that the client is actually sending unwrapped yaw values to the server

**Comment by jivvy** [comment]
I put a print statement in where the C2S packets are handled:
```rust
C2sPlayPacket::SetPlayerPositionAndRotation(p) => {
    println!("yaw: {}", p.yaw);
    if self.pending_teleports == 0 {
        self.position = p.position;
        self.yaw = wrap_yaw(p.yaw);
        self.pitch = p.pitch;

        self.events.push_back(ClientEvent::MovePositionAndRotation {
            position: p.position,
            yaw: p.yaw,
            pitch: p.pitch,
            on_ground: p.on_ground,
        });
    }
}
```
And interpreted this as that the client is actually sending unwrapped yaw values to the server

File: src/util.rs [hunk 1]
```diff
@@ -105,6 +105,15 @@ pub fn ray_box_intersect(ro: Vec3<f64>, rd: Vec3<f64>, bb: Aabb<f64>) -> Option<
     }
 }
 
+/// Takes a float and wraps it around -180..180 in a way that matches the
+/// client's interpretation of yaw values
+pub fn wrap_yaw(yaw: f32) -> f32 {
```
Comment (dyc3) [inline src/util.rs:110]: Do we need to make this public?

Comment (jivvy) [reply src/util.rs:110]: I'm not sure. 

The `Client` and `Entity` structs are public, so the `yaw` field they possess is also public.
If someone wanted to set the yaw, they would need the function.

Perhaps there's a better way of hiding `wrap_yaw` and apply it discreetly?

**Comment by rj00a** [comment]
Hmm I'm not sure this is an issue that should be fixed. If anything, I would expect all angles to be in `[0, 360)` instead of whatever Minecraft is doing. And since these are the actual values sent by the client, I don't think we should attempt to apply our own transformations.

**Comment by jivvy** [comment]
That's true. I'm not sure what kinds of applications would require a yaw fitting the vanilla client

**Comment by rj00a** [comment]
Thanks for opening a PR. Closing for the reasons discussed.

**Pull request closed without merging**